### PR TITLE
Firelock fixes

### PR DESCRIPTION
**Pull request by ElectroJr**
Fixes firelock pressure checks when adjacent to space.

### CODE REVIEW

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.API.cs [hunk 1]
```diff
@@ -108,7 +108,12 @@ public void InvalidateTile(EntityUid gridUid, Vector2i tile)
             RaiseLocalEvent(ref ev);
 
         // Default to a space mixture... This is a space game, after all!
-        return ev.Mixtures ?? new GasMixture?[tiles.Count];
+        ev.Mixtures ??= new GasMixture?[tiles.Count];
```
Comment (wrexbe) [inline Content.Server/Atmos/EntitySystems/AtmosphereSystem.API.cs:111]: Maybe it should just check if the event was handled here after raising it again, and then replace it to the space gas mixture without the null checks if the event wasn't handled.

**Review by wrexbe — CHANGES_REQUESTED** [review]

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.API.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Server/Atmos/EntitySystems/AtmosphereSystem.API.cs:111]: yeah that makes more sense, will do

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.Map.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ private void MapGetTileMixtures(EntityUid uid, MapAtmosphereComponent component,
 
         for (var i = 0; i < args.Tiles.Count; i++)
         {
-            args.Mixtures[i] = component.Mixture.Clone();
+            args.Mixtures[i] ??= component.Mixture.Clone();
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Atmos/EntitySystems/AtmosphereSystem.Map.cs:45]: Ah I missed the fact that it did that earlier. It should only be overwriting null entries in the list, and not overwriting the whole list

[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Atmos/EntitySystems/AtmosphereSystem.Map.cs:42]: > Not sure what the goal of this is, but should it be setting all of the mixtures to space gas before returning on component.Mixture == null?

I've now made it not handle the event, so the function that raised the event should do that now

**Review by wrexbe — APPROVED** [review]

**Pull request merged**